examples/chatgpt/gpt_actions_library/gpt_middleware_azure_function.ipynb

```diff
@@ -0,0 +1,494 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "12. Try to hit endpoint you created in Postman using those OAuth settings:\n",
+    "\n",
+    "    1. **Grant Type:** Authorization Code\n",
+    "\n",
+    "    2. **Auth URL**: [https://login.microsoftonline.com/](about:blank)`TENANT_ID`[/oauth2/v2.0/authorize](about:blank)\n",
+    "\n",
+    "    3. **Auth Token URL**: [https://login.microsoftonline.com/`TENANT_ID`/oauth2/v2.0/token](about:blank)\n",
+    "\n",
+    "    4. **Client ID:** `CLIENT_ID` from step 7 above\n",
+    "\n",
+    "    5. **Client secret:** `MICROSOFT_PROVIDER_AUTHENTICATION_SECRET `from step 11 above\n",
+    "\n",
+    "    6. **Scope**: `SCOPE` from step 5 above\n",
+    "\n",
+    "    7. **Client credentials**: Send client credentials in body\n",
+    "\n",
+    "13. You will need to click **Get New Access Token**, and then hit the endpoint you saved in step 10 above. If it was successful, you should get this response: `”This HTTP triggered function executed successfully. Pass a name in the query string or in the request body for a personalized response.”`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Part 5: Set up your Application on an Azure Function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This should be done separately and is specific to your app. See the [Sharepoint Cookbook]((https://cookbook.openai.com/examples/chatgpt/gpt_actions_library/gpt_action_sharepoint_doc)) for an example of that. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Part 6: Set up ChatGPT"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "14. Generate an OpenAPI spec for your endpoint. \n",
+    "\n",
+    "15. Paste that into the Actions section of a GPT, and choose OAuth as the authentication type. Fill out the OAuth settings the same way you did for Postman above. \n",
+    "\n",
+    "16. Once you save the action, you will see a callback URI at the bottom of the GPT configuration. Copy that URL, then go **back to your Function App in the Azure Portal**.\n",
+    "\n",
+    "17. Click on **Authentication** under **Settings**, then click on your Entra application.\n",
+    "\n",
+    "18. Once you are there, then click **Authentication** under the **Manage** section.\n",
+    "\n",
+    "19. Add a new Redirect URI under the **Web** section of that page, and paste in the Callback URI you got from step 16, then click Save. \n",
+    "\n",
+    "20. Test out the GPT and it should work as expected."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## ChatGPT Steps"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Custom GPT Instructions "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "*This is application specific.*"
```

Comment:
maybe another reference to the sharepoint instructions, linking to the specific header, as an example?